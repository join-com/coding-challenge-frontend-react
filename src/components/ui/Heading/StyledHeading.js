import React from "react";
import styled, { css } from "styled-components";
import { colors, fonts, fontSizes } from "../../../constants/styles";

const titles = {
  h1: css`
    font-size: ${fontSizes.extraLarge}rem;
    margin-bottom: ${fontSizes.extraLarge / 2}rem;
  `,

  h2: css`
    font-size: ${fontSizes.large * 1.2}rem;
    margin-bottom: ${(fontSizes.large * 1.2) / 2}rem;
  `,
  h3: css`
    font-size: ${fontSizes.large}rem;
    margin-bottom: ${fontSizes.large / 2}rem;
  `,
  h4: css`
    font-size: ${fontSizes.normal * 1.5}rem;
    margin-bottom: ${(fontSizes.normal * 1.5) / 2}rem;
  `,
  h5: css`
    font-size: ${fontSizes.normal * 1.2}rem;
    margin-bottom: ${(fontSizes.normal * 1.2) / 2}rem;
  `
};

export default styled(({ children, level, ...props }) => React.createElement(`h${level}`, props, children))`
  font-family: ${fonts.primary};
  font-weight: ${({ bold }) => (bold ? "bold" : 500)};
  line-height: 1.3;
  color: ${colors.dark};
  text-transform: ${({ uppercase }) => (uppercase ? "uppercase" : "none")};
  ${({ level }) => titles[`h${level}`]};
`;
