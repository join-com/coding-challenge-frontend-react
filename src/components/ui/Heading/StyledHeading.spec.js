import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import StyledHeading from "./StyledHeading";

describe("Styled StyledHeading component", () => {
  const defaultProps = {
    level: 1,
    uppercase: false,
    bold: false
  };

  const getWrapper = (testProps = {}) => {
    const props = { ...defaultProps, ...testProps };
    return renderer.create(<StyledHeading {...props} />).toJSON();
  };

  it("should match snapshot with different level prop", () => {
    for (let index = 1; index < 6; index++) {
      const wrapper = getWrapper({ level: index });
      expect(wrapper).toMatchSnapshot();
    }
  });

  it("should match snapshot with bold prop", () => {
    const wrapper = getWrapper({ bold: true });
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with uppercase prop", () => {
    const wrapper = getWrapper({ uppercase: true });
    expect(wrapper).toMatchSnapshot();
  });
});
