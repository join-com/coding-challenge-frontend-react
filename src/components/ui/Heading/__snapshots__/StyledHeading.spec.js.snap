// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled StyledHeading component should match snapshot with bold prop 1`] = `
.c0 {
  font-family: 'Oswald',sans-serif;
  font-weight: bold;
  line-height: 1.3;
  color: #1D232B;
  text-transform: none;
  font-size: 3rem;
  margin-bottom: 1.5rem;
}

<h1
  bold={true}
  className="c0"
  uppercase={false}
/>
`;

exports[`Styled StyledHeading component should match snapshot with different level prop 1`] = `
.c0 {
  font-family: 'Oswald',sans-serif;
  font-weight: 500;
  line-height: 1.3;
  color: #1D232B;
  text-transform: none;
  font-size: 3rem;
  margin-bottom: 1.5rem;
}

<h1
  bold={false}
  className="c0"
  uppercase={false}
/>
`;

exports[`Styled StyledHeading component should match snapshot with different level prop 2`] = `
.c0 {
  font-family: 'Oswald',sans-serif;
  font-weight: 500;
  line-height: 1.3;
  color: #1D232B;
  text-transform: none;
  font-size: 1.7999999999999998rem;
  margin-bottom: 0.8999999999999999rem;
}

<h2
  bold={false}
  className="c0"
  uppercase={false}
/>
`;

exports[`Styled StyledHeading component should match snapshot with different level prop 3`] = `
.c0 {
  font-family: 'Oswald',sans-serif;
  font-weight: 500;
  line-height: 1.3;
  color: #1D232B;
  text-transform: none;
  font-size: 1.5rem;
  margin-bottom: 0.75rem;
}

<h3
  bold={false}
  className="c0"
  uppercase={false}
/>
`;

exports[`Styled StyledHeading component should match snapshot with different level prop 4`] = `
.c0 {
  font-family: 'Oswald',sans-serif;
  font-weight: 500;
  line-height: 1.3;
  color: #1D232B;
  text-transform: none;
  font-size: 1.5rem;
  margin-bottom: 0.75rem;
}

<h4
  bold={false}
  className="c0"
  uppercase={false}
/>
`;

exports[`Styled StyledHeading component should match snapshot with different level prop 5`] = `
.c0 {
  font-family: 'Oswald',sans-serif;
  font-weight: 500;
  line-height: 1.3;
  color: #1D232B;
  text-transform: none;
  font-size: 1.2rem;
  margin-bottom: 0.6rem;
}

<h5
  bold={false}
  className="c0"
  uppercase={false}
/>
`;

exports[`Styled StyledHeading component should match snapshot with uppercase prop 1`] = `
.c0 {
  font-family: 'Oswald',sans-serif;
  font-weight: 500;
  line-height: 1.3;
  color: #1D232B;
  text-transform: uppercase;
  font-size: 3rem;
  margin-bottom: 1.5rem;
}

<h1
  bold={false}
  className="c0"
  uppercase={true}
/>
`;
